server/tserver/src/main/java/org/apache/accumulo/tserver/ScanServer.java

```diff
@@ -414,13 +426,72 @@ private TabletMetadata getTabletMetadata(KeyExtent extent) {
     }
   }
 
+  protected boolean insertScanReferences(TabletMetadata tm) {
+    final String serverAddress = clientAddress.toString();
+    final String clientAddress = TServerUtils.clientAddress.get();
+    final Set<ScanServerRefTabletFile> refsToAdd = new HashSet<>();
+    tm.getFiles().forEach(f -> {
+      AtomicInteger count = this.allReservedFiles.computeIfAbsent(f, x -> new AtomicInteger(0));
+      if (count.get() == 0) {
```

Comment:
Thinking referencing counting will be really hard to get correct.  Also thinking that even when no scan us currently using files that we should probably leave the scan server references in the metadata table for a bit to avoid the expense of inserting them again.  Not sure how to do it w/o looking more at the tablet server code, but thinking it would be nice to only remove scan server refs that meet the following criteria.

 * Not in use by any scan in the scan server
 * Has not been used for a few minutes.

So thinking we could periodically look for scan refs that are not currently in use and have not been used recently and delete them.

If we are constantly reading and writing to the metadata table for each scan on a scan server, it will make scans very expensive.